chain/protocol_state.go

```diff
@@ -61,6 +63,16 @@ func (ps *ProtocolState) Active() (uint64, flow.Identifier, flow.StateCommitment
 	return ps.height, ps.blockID, ps.commit
 }
 
+func (ps *ProtocolState) Events() ([]flow.Event, error) {
+	var events []flow.Event
+	err := operation.LookupEventsByBlockID(ps.blockID, &events)(ps.state.NewTransaction(false))
```

Comment:
I wonder what happens for blocks where there are no events at all. Could you double-check in the Flow Go source code? In case it errors, we have to use `errors.Is(err, storage.ErrNotFound)` in the consumer of this function to catch that edge case and simply index an empty events slice there.

Reply:
I checked it already (since in my case, there are no events at all in the blocks that I have for some reason, this is probably an issue btw I forgot to mention it this morning :p) and it didn't seem to error.

I can double check if you'd like.

Reply:
It seems like indeed it cannot error because of no values being there. See https://github.com/onflow/flow-go/blob/master/storage/badger/operation/common.go#L327-L381